### PR TITLE
Update copy_on_model_validation tests

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @Kludex

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -1144,36 +1144,9 @@ class SubUser(User):
     assert not hasattr(t.user, 'sins')
 
 
-@pytest.mark.skip(reason='not implemented')
-@pytest.mark.parametrize('comv_value', [True, False])
-def test_copy_on_model_validation_warning(comv_value):
```
Comment (hramezani) [inline tests/test_main.py:1149]: This replaced by:

- (test_revalidate_instances_never)[https://github.com/pydantic/pydantic/blob/ca8069811a30c271c9ee0b36d455c9c0adb458a0/tests/test_main.py#L1070]
- (test_revalidate_instances_sub_instances)[https://github.com/pydantic/pydantic/blob/ca8069811a30c271c9ee0b36d455c9c0adb458a0/tests/test_main.py#L1095]
- (test_revalidate_instances_always)[https://github.com/pydantic/pydantic/blob/ca8069811a30c271c9ee0b36d455c9c0adb458a0/tests/test_main.py#L1121]

**Comment by hramezani** [comment]
please review

File: tests/test_main.py [hunk 2]
```diff
@@ -1144,36 +1144,9 @@ class SubUser(User):
...
-    else:
-        assert t.user is my_user
-    assert t.user.hobbies is my_user.hobbies
-
-
-@pytest.mark.skip(reason='not implemented')
-def test_validation_deep_copy():
-    """By default, Config.copy_on_model_validation should do a deep copy"""
-
+def test_revalidate_instances_always1():
```
Comment (Kludex) [inline tests/test_main.py:1147]: Hmm... Can we replace the `1` by a more meaningful name? 🤔

Comment (hramezani) [reply tests/test_main.py]: Ah, just renamed it

**Review by Kludex — APPROVED** [review]

**Pull request merged**